src/lib/components/ThemeToggle/index.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Element that allows a user to toggle the current theme.
+ * @extends {HTMLElement}
+ * @final
+ */
+class ThemeToggle extends HTMLElement {
+  constructor() {
+    super();
+
+    this.STORAGE_KEY = 'user-color-scheme';
+    this.COLOR_MODE_KEY = '--color-mode';
+  }
+
+  connectedCallback() {
+    // Allows --flow-space to work and we
+    // only want that to happen when it's
+    // wired up
+    this.style.display = 'block';
```

Comment:
I left a comment down in the render function which I think is related to this. I was wondering if we can go ahead and have all of the html and CSS for the element stamped out already, and in connectedCallback we only do the afterRender() stuff.